test/scalar_tests/src/float/test_float_convert.c

```diff
@@ -172,9 +165,6 @@ TEST(float_convert, f32_to_float_s32)
 
     float_s32_t actual = f32_to_float_s32(x);
 
-    if(x != 0.0)
-      TEST_ASSERT_LESS_OR_EQUAL(1, HR_S32(actual.mant));
-
```

Comment:
`f32_to_float_s32` used to be a high-level `math.h` implementation. Now that this tests assembly as well, we can't have this requirement as the asm impl always returns a headroom of 7. The accuracy of the output value is the same tho.